lib/view_component/isolated_i18n.rb

```diff
@@ -0,0 +1,44 @@
+require 'set'
+require 'action_view/helpers/translation_helper'
+
+module ViewComponent::IsolatedI18n
+  def self.included(base)
+    base.class_attribute :isolated_i18n_backend, instance_writer: false, instance_predicate: false
+    base.extend ClassMethods
+  end
+
+  Cache = Set.new
+
+  def translate(key = nil, locale: nil, **options) # TODO deprecate :raise
+    locale ||= ::I18n.locale
+    backend = isolated_i18n_backend
+
+    result = catch(:exception) do
+      if key.is_a?(Array)
+        key.map { |k| backend.translate(locale, k, options) }
+      else
+        backend.translate(locale, key, options)
+      end
+    end
+
+    if result.is_a?(I18n::MissingTranslation)
+      handle_exception(false, result, locale, key, options)
+    else
+      result
+    end
+  end
+  alias :t :translate
+
+  module ClassMethods
+    def compile(*,**)
```

Comment:
It's currently private and experimental, but I think `self._after_compile` is another potential place to hook into the compile process. https://github.com/github/view_component/blob/main/lib/view_component/compiler.rb#L48

Reply:
I saw that (also the experimental warning) but finally decided that using it would put the burden of calling super on anyone that wanted to use it in their components. Also being `IsolatedI18n` part of ViewComponent seemed ok to hook into a core method.

The assumption is of course that this is meant to be used in the final app, probably inside the `ApplicationComponent`, if that is not the case I'm happy to move there the implementation.